pymc/logprob/order.py

```diff
@@ -0,0 +1,114 @@
+#   in the Software without restriction, including without limitation the rights
+#   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#   copies of the Software, and to permit persons to whom the Software is
+#   furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in all
+#   copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#   SOFTWARE.
+
+from typing import List, Optional
+
+from pytensor.graph.basic import Node
+from pytensor.graph.fg import FunctionGraph
+from pytensor.graph.rewriting.basic import node_rewriter
+from pytensor.tensor.math import MaxAndArgmax
+
+from pymc.logprob.abstract import MeasurableVariable, _logcdf, _logprob
+from pymc.logprob.rewriting import measurable_ir_rewrites_db
+
+
+class MeasurableMax(MaxAndArgmax):
+    """A placeholder used to specify a log-likelihood for a clipped RV sub-graph."""
+
+
+MeasurableVariable.register(MeasurableMax)
+
+
+@node_rewriter([MaxAndArgmax])
+def find_measurable_max(fgraph: FunctionGraph, node: Node) -> Optional[List[MeasurableMax]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableMax):
+        return None  # pragma: no cover
+
+    base_var = node.inputs[0]
+
+    if base_var.owner.inputs[3].type.ndim != 0:
+        return None
+
+    if not rv_map_feature.request_measurable(node.inputs):
+        return None
+
+    axis = node.op.axis
+    measurable_max = MeasurableMax(list(axis))
+    max_rv_node = measurable_max.make_node(base_var)
+    max_rv = max_rv_node.outputs
+
+    return max_rv
+
+
+measurable_ir_rewrites_db.register(
+    "find_measurable_max",
+    find_measurable_max,
+    "basic",
+    "max",
+)
+
+
+@_logprob.register(MeasurableMax)
+def max_logprob(op, values, base_rv, **kwargs):
+    (value,) = values
+
+    base_rv_op = base_rv.owner.op
+    base_rv_inputs = base_rv.owner.inputs
+
+    logprob = _logprob(base_rv_op, (value,), *base_rv_inputs, **kwargs)
+    logcdf = _logcdf(base_rv_op, value, *base_rv_inputs, **kwargs)
+
+    if base_rv_op.name:
+        logprob.name = f"{base_rv_op}_logprob"
+        logcdf.name = f"{base_rv_op}_logcdf"
+
+    size_var = base_rv.owner.inputs[1]
+    string_size = str(size_var)
+    for b in (0, len(string_size) - 1):
+        if string_size[b] == "}":
+            a = string_size[b - 1]
```

Comment:
This should just use `value.size`. In general you don't need to worry about having constants, it's fine for a model to have a `size` that's dynamic and perfectly valid to create PyTensor logprob expressions using non-constant size variables.